src/async-wrap.cc

```diff
@@ -76,6 +76,8 @@ static void Initialize(Handle<Object> target,
 Handle<Value> AsyncWrap::MakeCallback(const Handle<Function> cb,
                                       int argc,
                                       Handle<Value>* argv) {
+  if (!env()->CanCallIntoJs())
+    return Undefined(env()->isolate());
```

Comment:
I just see that `CanCallIntoJs()` calls `is_main_instance()` which is simply `worker_context_ == nullptr`.